charts/ai-service/templates/deployment.yaml

```diff
@@ -158,25 +135,14 @@ spec:
       {{- with .Values.volumes }}
         {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- if .Values.pvc.enabled }}
-        - name: {{ .Values.pvc.name | default $fullName }}
+      {{- if .Values.artifactsCache.enabled }}
+        - name: artifacts-volume
           persistentVolumeClaim:
-            claimName: {{ .Values.pvc.name | default $fullName }}
-      {{- end }}
-      {{- if .Values.deployment.initContainers }}
-      {{- range .Values.deployment.initContainers }}
-      {{- if .volumeMounts }}
-      {{- range .volumeMounts }}
-       {{- if eq .emptyDir true }}
-        - name: {{ .name }}
-          emptyDir: {}
-       {{- end }}
-      {{- end }}
-      {{- end }}   
-      {{- end }}
-      {{- end }}
-      {{- with .Values.volumes }}
-        {{- toYaml . | nindent 8 }}
+            claimName: {{ include "common.fullname" . }}-artifacts
+        - name: download-script
+          configMap:
```

Comment:
This configMap seems neat, I'm not sure can you explain what we want to achieve?